Chnage Recepie component

The recepie button, show recepie for the old meal:
After 13:00 the dashboard begins to show meal for the next day, so the recepie should do the same, my guess is that it updates based on the day and not time.

#15 [builder 6/6] RUN npm run build
#15 0.425 
#15 0.425 > client@0.1.0 build
#15 0.425 > react-scripts build
#15 0.425 
#15 1.731 Creating an optimized production build...
#15 21.29 Failed to compile.
#15 21.29 
#15 21.29 TS2345: Argument of type '{ date: string; day: string; foodName: string; correctedFoodName?: string | undefined; description?: string | undefined; recipe?: string | undefined; foodContents: string[]; weekNumber: number; image?: { ...; } | undefined; }' is not assignable to parameter of type 'SetStateAction<MenuItem | null>'.
#15 21.29     21 |                 const filteredData = filterMenu(data);
#15 21.29     22 |                 if (filteredData && filteredData.length > 0) {
#15 21.29   > 23 |                     setMenuItem(filteredData[0]);
#15 21.29        |                                 ^^^^^^^^^^^^^^^
#15 21.29     24 |                 }
#15 21.29     25 |             } catch (error) {
#15 21.29     26 |                 console.error('Failed to load menu item:', error);
#15 21.29 
#15 21.29 
#15 ERROR: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
------
 > [builder 6/6] RUN npm run build:
21.29 TS2345: Argument of type '{ date: string; day: string; foodName: string; correctedFoodName?: string | undefined; description?: string | undefined; recipe?: string | undefined; foodContents: string[]; weekNumber: number; image?: { ...; } | undefined; }' is not assignable to parameter of type 'SetStateAction<MenuItem | null>'.
21.29     21 |                 const filteredData = filterMenu(data);
21.29     22 |                 if (filteredData && filteredData.length > 0) {
21.29   > 23 |                     setMenuItem(filteredData[0]);
21.29        |                                 ^^^^^^^^^^^^^^^
21.29     24 |                 }
21.29     25 |             } catch (error) {
21.29     26 |                 console.error('Failed to load menu item:', error);
21.29 
21.29 
------
Dockerfile:7
--------------------
   5 |     RUN npm install
   6 |     COPY . .
   7 | >>> RUN npm run build
   8 |     
   9 |     #Stage 2
--------------------
ERROR: failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
Error: Process completed with exit code 1.

--- a/src/index.tsx
+++ b/src/index.tsx
@@ -19,6 +19,7 @@ import MenuEditor from './routes/editor/MenuEditor';
 import WindowManager from './routes/editor/WindowManager';
 import ProgressAdoEditor from './routes/editor/ProgressAdoEditor';
 import RecipeView from './routes/RecipeView';
+import RecipeModal from './routes/RecipeModal';
 
 const router = createBrowserRouter([
     {
@@ -58,6 +59,10 @@ const router = createBrowserRouter([
             {
                 path: 'recipe-view',
                 element: <RecipeView />
+            },
+            {
+                path: 'recipe-modal',
+                element: <RecipeModal />
             },
             {
                 path: 'edit',

--- a/src/routes/Dashboard.tsx
+++ b/src/routes/Dashboard.tsx
@@ -273,7 +273,7 @@ function Dashboard() {
                                     {menus.length > 0 && (menus[0].recipe || menus[0].needsRecipeGeneration) && (
                                         <div style={{ padding: '10px', borderTop: '1px solid #ddd', display: 'flex', justifyContent: 'center' }}>
                                             <button 
-                                                onClick={() => navigate('/recipe-view')}
+                                                onClick={() => navigate('/recipe-modal')}
                                                 disabled={menus.length === 0 || menus[0].needsRecipeGeneration}
                                                 style={{ 
                                                     padding: '5px 10px',

--- a/src/routes/RecipeModal.tsx
+++ b/src/routes/RecipeModal.tsx
@@ -20,7 +20,8 @@ function RecipeModal() {
                 // Apply the same time-based filtering that the dashboard uses
                 const filteredData = filterMenu(data);
                 if (filteredData && filteredData.length > 0) {
-                    setMenuItem(filteredData[0]);
+                    // Explicitly cast the menu item to ensure type compatibility
+                    setMenuItem(filteredData[0] as MenuItem);
                 }
             } catch (error) {
                 console.error('Failed to load menu item:', error);

--- a/src/routes/RecipeModal.tsx
+++ b/src/routes/RecipeModal.tsx
@@ -1,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Box, Typography } from '@mui/material';
+import Window from '../Window';
+import win7bg from '../images/win7bg.jpg';
+import { MenuItem } from '../apiClient/apiClient';
+import { filterMenu } from '../MenuItem';
+import Markdown from 'react-markdown'
+
+function RecipeModal() {
+    const [menuItem, setMenuItem] = useState<MenuItem | null>(null);
+    const [loading, setLoading] = useState(true);
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        const fetchCurrentMenu = async () => {
+            try {
+                const response = await fetch('/api/menu');
+                const data = await response.json();
+                // Apply the same time-based filtering that the dashboard uses
+                const filteredData = filterMenu(data);
+                if (filteredData && filteredData.length > 0) {
+                    setMenuItem(filteredData[0]);
+                }
+            } catch (error) {
+                console.error('Failed to load menu item:', error);
+            } finally {
+                setLoading(false);
+            }
+        };
+
+        fetchCurrentMenu();
+    }, []);
+
+    return (
+        <div style={{position: 'absolute', top: 0, left: 0, bottom: 0, right: 0, backgroundImage: `url(${win7bg})`}}>
+            <Window 
+                title={'Opskrift - ' + (menuItem?.foodDisplayName || 'Indlæser...')} 
+                maximized={true} 
+                fullscreen={true} 
+                sx={{
+                    margin: '24px',
+                    height: 'calc(100vh - 48px)',
+                    top: 0,
+                    position: 'absolute',
+                    width: 'calc(100vw - 48px)'
+                }} 
+                maximizeCallback={() => navigate(-1)}
+            >
+                {loading ? (
+                    <Typography variant="body1" sx={{ padding: 2 }}>Indlæser opskrift...</Typography>
+                ) : menuItem ? (
+                    <>
+                        {menuItem.recipe ? (
+                            <Box sx={{ padding: 2, overflowY: 'auto', maxHeight: 'calc(100vh - 120px)' }}>
+                                <Markdown>{menuItem.recipe}</Markdown>
+                            </Box>
+                        ) : menuItem.needsRecipeGeneration ? (
+                            <Typography variant="body1" sx={{ padding: 2 }}>
+                                Opskrift er under udarbejdelse...
+                            </Typography>
+                        ) : (
+                            <Typography variant="body1" sx={{ padding: 2 }}>
+                                Ingen opskrift tilgængelig for denne ret.
+                            </Typography>
+                        )}
+                    </>
+                ) : (
+                    <Typography variant="body1" sx={{ padding: 2 }}>Ingen menu fundet.</Typography>
+                )}
+            </Window>
+        </div>
+    );
+}
+
+export default RecipeModal;

--- a/src/routes/RecipeView.tsx
+++ b/src/routes/RecipeView.tsx
@@ -4,6 +4,7 @@ import { Box, Button, Typography } from '@mui/material';
 import Window from '../Window';
 import win7bg from '../images/win7bg.jpg';
 import { MenuItem } from '../apiClient/apiClient';
+import { filterMenu } from '../MenuItem';
 import Markdown from 'react-markdown'
 
 function RecipeView() {
@@ -16,8 +17,10 @@ function RecipeView() {
             try {
                 const response = await fetch('/api/menu');
                 const data = await response.json();
-                if (data && data.length > 0) {
-                    setMenuItem(data[0]);
+                // Apply the same time-based filtering that the dashboard uses
+                const filteredData = filterMenu(data);
+                if (filteredData && filteredData.length > 0) {
+                    setMenuItem(filteredData[0]);
                 }
             } catch (error) {
                 console.error('Failed to load menu item:', error);

--- a/src/routes/RecipeView.tsx
+++ b/src/routes/RecipeView.tsx
@@ -20,7 +20,8 @@ function RecipeView() {
                 // Apply the same time-based filtering that the dashboard uses
                 const filteredData = filterMenu(data);
                 if (filteredData && filteredData.length > 0) {
-                    setMenuItem(filteredData[0]);
+                    // Explicitly cast the menu item to ensure type compatibility
+                    setMenuItem(filteredData[0] as MenuItem);
                 }
             } catch (error) {
                 console.error('Failed to load menu item:', error);

